apps/web/src/hooks/integrations/useHasActiveIntegrations.ts

```diff
@@ -0,0 +1,65 @@
+import { ChannelTypeEnum, EmailProviderIdEnum, SmsProviderIdEnum } from '@novu/shared';
+import { useMemo } from 'react';
+import { useEnvController } from '../useEnvController';
+import { useActiveIntegrations } from './useActiveIntegrations';
+import { useIntegrationLimit } from './useIntegrationLimit';
+type UseHasActiveIntegrationsProps = {
+  filterByEnv?: boolean;
+  channelType: ChannelTypeEnum;
+};
+
+export function useHasActiveIntegrations({ filterByEnv = true, channelType }: UseHasActiveIntegrationsProps) {
+  const { integrations } = useActiveIntegrations();
+  const { environment } = useEnvController();
+  const { isLimitReached: isEmailLimitReached } = useIntegrationLimit(ChannelTypeEnum.EMAIL);
+  const { isLimitReached: isSmsLimitReached } = useIntegrationLimit(ChannelTypeEnum.SMS);
+
+  const isChannelStep = useMemo(() => {
+    return [
+      ChannelTypeEnum.IN_APP,
+      ChannelTypeEnum.EMAIL,
+      ChannelTypeEnum.PUSH,
+      ChannelTypeEnum.SMS,
+      ChannelTypeEnum.CHAT,
+    ].includes(channelType);
+  }, [channelType]);
+
+  const activeIntegrationsByEnv = useMemo(() => {
+    if (filterByEnv) {
+      return integrations?.filter((integration) => integration._environmentId === environment?._id);
```

Comment:
missing check `integration.active && ...`

Reply:
integrations here is received from useActiveIntegrations hook, all the integrations will have active = true

Reply:
ahh right missed it :)